### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by popovychyb**

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -3,13 +3,87 @@
...
+            }
+        }
         return null;
     }
+
+    private static class Entry<K, V> {
+        private K key;
+        private V value;
+
+        public Entry(K key, V value) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:44]: do we need public constructor for Entry here?

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,13 +3,87 @@
...
+            this.key = key;
+            this.value = value;
+        }
+
+        public K getKey() {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:49]: same here with access modifier

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,13 +3,87 @@
...
+        }
     }
 
     @Override
     public V get(K key) {
+        if (entries[0] == null) {
+            return null;
+        }
+        for (int i = 0; i < entries.length; i++) {
+            if (key == entries[i].getKey() || (key != null && key.equals(entries[i].getKey()))) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:33]: what happens with entries[i].getKey() if it's not initialized ? 
for example we have initialize only entries[0] and entries[1] == null

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,13 +3,87 @@
...
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            Entry<?, ?> entry = (Entry<?, ?>) o;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:74]: make as in previous task

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 1 above)
Comment (popovychyb) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: Done. Yes, private is better here. Thank you!

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 2 above)
Comment (popovychyb) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: Done. Thank you!

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 3 above)
Comment (popovychyb) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: Done. Fixed checking for not initialized element of the array. Thank you!

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 4 above)
Comment (popovychyb) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: Done. Thank you!

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 5]
```diff
@@ -3,13 +3,87 @@
...
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            Entry<K, V> entry = (Entry) o;
+
+            if (key != null ? !key.equals(entry.key) : entry.key != null) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:76]: optimize this check to smt like this 
```
return (entry.getKey() == key
                || (entry.getKey != null && entry.getKey().equals(key)))
                && (entry.getValue() == value
                || (entry.getValue() != null && entry.getValue().equals(value)));
```

Comment (popovychyb) [reply src/main/java/core/basesyntax/impl/StorageImpl.java]: Done. Thank you!

**Pull request closed without merging**